ocamltest/run_win32.c

```diff
@@ -175,18 +175,40 @@ static LPVOID prepare_environment(WCHAR **localenv)
   }
   r = env;
   p = process_env;
+  /* Copy process_env to env only if the given names are not in localenv */
   while (*p != L'\0') {
+    wchar_t *pos_eq = wcschr(p, L'=');
+    int copy = 1;
     l = wcslen(p) + 1; /* also count terminating '\0' */
-    memcpy(r, p, l * sizeof(WCHAR));
+    /* Temporarily change the = to \0 for wcscmp */
+    *pos_eq = L'\0';
+    q = localenv;
+    while (*q != NULL) {
+      wchar_t *pos_eq2 = wcschr(*q, L'=');
+      /* Compare this name in localenv with the current one in processenv */
+      if (pos_eq2) *pos_eq2 = L'\0';
+      if (!wcscmp(*q, p)) copy = 0;
+      if (pos_eq2) *pos_eq2 = L'=';
+      q++;
+    }
+    if (copy) {
+      /* This name is not marked for deletion/update in localenv, so copy */
+      *pos_eq = L'=';
```

Comment:
I'd move this line up outside the `if`. The assigment to `*pos_eq` is supposed to be temporary, so it should be undone in all cases.

Reply:
A good point, and changed.